crates/sui-graphql-rpc/src/extensions/query_limits_checker.rs

```diff
@@ -328,3 +412,63 @@ fn check_directives(directives: &[Positioned<Directive>]) -> ServerResult<()> {
     }
     Ok(())
 }
+/// Given a node, estimate the number of output nodes it will produce.
+fn estimate_output_nodes_for_curr_node(
+    f: &Positioned<Field>,
+    variables: &Variables,
+    limits: &Limits,
```

Comment:
nit: It looks like you only use `default_page_size` out of this -- consider making that the parameter?